datalab/datalab_session/analysis/get_tif.py

```diff
@@ -0,0 +1,24 @@
+from datalab.datalab_session.util import create_tif, get_fits, add_file_to_bucket, key_exists, get_s3_url
+
+def get_tif(input: dict):
+  """
+    Checks bucket for tif file and returns the url
+    if the file doesn't exist, generates a new tif file
+    input: dict
+      basename: str
+      source: str
+  """
+
+  basename = input["basename"]
+  file_key = f'{basename}/{basename}.tif'
```

Comment:
Are all the files associated with a file stored in the `{basename}/` folder but then also have the `{basename}.extension` filename too? I guess the max items we would generate in one of those folders is the .fits, .tiff, and .jpeg right now?

Reply:
yeah thats how it is currently, and the most it has are .fits, .tiff, .jpeg x 2 (with the -large or -small modifier). I also added this line for the future `bucket_key = f'{cache_key}/{cache_key}-{index}' if index else f'{cache_key}/{cache_key}'` but so far no operation generates more than one file. And also for the case of the tif then the basename would include the `-n`.